### PR TITLE
UI metadata and server needed to query the background

### PR DESCRIPTION
**Pull request by sbrunner**
Related to https://github.com/camptocamp/ngeo/issues/1238

### CODE REVIEW

**Comment by sbrunner** [comment]
Wedeay to be reviewed, @fredj can you do it?

File: c2cgeoportal/views/entry.py [hunk 1]
```diff
@@ -562,28 +566,31 @@ def _is_internal_wms(self, layer):
...
-    def _get_layer_identifiers(self, group):
-        """Recurse on all children to get unique identifier for each child."""
-        identifier = []
+    def _get_ogc_servers(self, group, depth=1):
+        """ Recurse on all children to get unique identifier for each child. """
+        ogc_servers = set()
+
+        # escape loop
+        if depth > 30:
+            print("Error: too many recursions with group '%s'" % group.name)
```
Comment (fredj) [inline c2cgeoportal/views/entry.py]: `log.error` ?

File: doc/integrator/ngeo.rst [hunk 2]
```diff
@@ -99,3 +99,14 @@ Background layers
 
 The background layers are configured in the database, with the layer group named
 **background** (by default).
+
+WMTS Layers
+===========
+
+To make the WMTS queryabe you should add those UI Metadata:
```
Comment (fredj) [inline doc/integrator/ngeo.rst]: queryable instead of queryabe

**Comment by fredj** [comment]
LGTM

**Comment by sbrunner** [comment]
Thanks @fredj

File: c2cgeoportal/views/entry.py [hunk 3]
```diff
@@ -1305,6 +1317,27 @@ def themes(self):
...
+                    server_ogc.url_wfs, self.request,
+                    default=url, errors=all_errors
+                )
+                result["serversOGC"][server_ogc.name] = {
+                    "url": url,
+                    "urlWfs": url_wfs,
+                    "type": server_ogc.type,
+                    "imageType": server_ogc.image_type,
+                    "auth": server_ogc.auth,
+                    "wtsSupport": server_ogc.wfs_support,
```
Comment (tsauerwein) [inline c2cgeoportal/views/entry.py]: This should be `wfsSupport`.

File: c2cgeoportal/tests/functional/test_themes.py [hunk 4]
```diff
@@ -473,6 +473,77 @@ def test_metadata(self):
...
         )
 
+    def test_server_ogc(self):
+        entry = self._create_entry_obj(params={
+            "version": "2",
+            "catalogue": "true",
+        })
+        themes = entry.themes()
+        self.assertEquals(self._get_filtered_errors(themes), set())
+        print(themes)
```
Comment (tsauerwein) [inline c2cgeoportal/tests/functional/test_themes.py]: forgotten debug statement

**Comment by fredj** [comment]
@tsauerwein PR to come

File: c2cgeoportal/views/entry.py (hunk 3 above)
Comment (sbrunner) [inline c2cgeoportal/views/entry.py]: oups, thanks :-)

**Pull request merged**